Draft docs for AI onboarding (#5987)
elastic/docs-content: solutions/elasticsearch-solution-project/ai-onboarding.md

---
navigation_title: AI onboarding
applies_to:
  stack: ga 9.4+
  serverless: ga
products:
  - id: cloud-serverless
  - id: elasticsearch
  - id: kibana
---

> # AI onboarding

Reviewer: ```suggestion
# AI onboarding in {{es}} projects
```

> **AI onboarding** is an AI-powered assistant that guides you from an initial idea for your use case to a working search experience through a conversational, use-case-driven flow.

Reviewer: ```suggestion
**AI onboarding** is a skill you can invoke in your IDE, CLI, or directly within your Elastic deployment using [](/explore-analyze/ai-features/elastic-agent-builder.md). This skill enables an AI agent to guide you from an initial idea for your use case to a working search experience, using natural language.
```

Feel free to change the verbiage, just tries to explain a little more concretely how it works, and links to agent builder.

> **AI onboarding** is a skill you can invoke in your IDE, CLI, or directly within your Elastic deployment using [](/explore-analyze/ai-features/elastic-agent-builder.md). This skill enables an AI agent to guide you from an initial idea for your use case to a working search experience, using natural language.

Reviewer 2: should there be a value in `[](/explore-analyze/ai-features/elastic-agent-builder.md)` or does it grab it from those docs automatically?

Reviewer: yep it grabs it automatically ✨

> The assistant asks what you’re building, understands your data, recommends the right approach, and generates a working implementation, rather than requiring you to read documentation or create templates.

Reviewer: ```suggestion
The agent asks what you’re building, understands your data, recommends the right approach, and generates a working implementation. This helps you get started immediately without reading documentation or creating templates.
```

nit to remove the overloaded term "assistant"

- **Avoid common pitfalls**: the assistant explains decisions and validates key steps

AI onboarding uses a guided, conversational flow that adapts to your use case.
It starts by understanding your intent, then analyzes your data and requirements.
Based on this, it recommends an appropriate search approach and guides you through implementation, including mappings, indexing, and queries.

## Get started with AI onboarding

When you start a new {{es}} project, the **Getting started** page opens. To access it later, use the **Getting started** menu.

:::{image} /solutions/images/ai-onboarding-paths.png
:alt: AI onboarding paths
:screenshot:
:::

You can choose between two onboarding paths:

### Use your IDE or CLI

Install Elastic onboarding skills in your preferred coding assistant (for example, Cursor or Claude Code):

1. Click **Copy prompt**.
2. Copy the generated prompt.

  :::{image} /solutions/images/ai-onboarding-copy-prompt.png
  :alt: AI onboarding prompt
  :screenshot:
  :::
3. Paste the prompt into your coding agent to install the onboarding skills.

Once installed, your coding agent can guide you through the onboarding flow.

### Use Elastic Agent

> Start onboarding directly in Elastic:

Reviewer: ```suggestion
Start onboarding directly in Elastic using the Agent Builder chat:
```

1. Click **Open Elastic Agent**.
2. A chat interface opens.

>   :::{image} /solutions/images/ai-onboarding-chat.png

Reviewer: You can also invoke the skill directly, not sure which one you'd prefer to emphasize

See https://www.elastic.co/docs/explore-analyze/ai-features/agent-builder/skills#how-skills-are-invoked

<img width="863" height="288" alt="Image" src="https://github.com/user-attachments/assets/44dd7ebb-5f44-4765-8637-c81593b75140" />

> ### Use Elastic Agent

Reviewer: ```suggestion
### Use Elastic AI Agent
```

> Start onboarding directly in Elastic using the Agent Builder chat:

Reviewer: ```suggestion
You can start onboarding directly in Elastic using the [Agent Builder](/explore-analyze/ai-features/elastic-agent-builder.md) chat. From the {{es}} getting started page in the UI:
```

> 1. Click **Open Elastic Agent**.

Reviewer: ```suggestion
1. Click **Open Elastic AI Agent**.
```

currently a typo in the UI, which will be fixed to avoid confusion with Elastic Agent

(see [Slack](https://elastic.slack.com/archives/C05E3NSDLS2/p1776984147426489) for context)

  :alt: AI onboarding chat
  :screenshot:
  :::
3. Describe what you want to build.

> The assistant asks about your use case and data, recommends an appropriate search approach, guides you through mappings and setup, and generates a working implementation.

Reviewer: ```suggestion
The agent asks about your use case and data, recommends an appropriate search approach, guides you through mappings and setup, and generates a working implementation.
```

:::{image} /solutions/images/ai-onboarding-paths.png
:alt: AI onboarding paths
:screenshot:
:::

You can choose between two onboarding paths:

### Use your IDE or CLI

Install Elastic onboarding skills in your preferred coding assistant (for example, Cursor or Claude Code):

1. Click **Copy prompt**.
2. Copy the generated prompt.

  :::{image} /solutions/images/ai-onboarding-copy-prompt.png
  :alt: AI onboarding prompt
  :screenshot:
  :::
3. Paste the prompt into your coding agent to install the onboarding skills.

Once installed, your coding agent can guide you through the onboarding flow.

### Use Elastic AI Agent

You can start onboarding directly in Elastic using the [Agent Builder](/explore-analyze/ai-features/elastic-agent-builder.md) chat. From the {{es}} getting started page in the UI:

1. Click **Open Elastic AI Agent**.
2. A chat interface opens.

  :::{image} /solutions/images/ai-onboarding-chat.png
  :alt: AI onboarding chat
  :screenshot:
  :::
3. Describe what you want to build.

The agent asks about your use case and data, recommends an appropriate search approach, guides you through mappings and setup, and generates a working implementation.

> You can also [invoke the onboarding skill directly](/explore-analyze/ai-features/agent-builder/skills#how-skills-are-invoked):

Author: ```suggestion
You can also [invoke the onboarding skill directly](/explore-analyze/ai-features/agent-builder/skills.md#how-skills-are-invoked):
```
